### ISSUE TITLE
Full Accessibility & 508 Office Audit Disability Claims

### ISSUE THREAD

**Issue by luke-gcio** [department-of-veterans-affairs/va.gov-team#9081]
##Note
Luke will update but created ticket for now to make sure we are tracking this.

## Steps to complete the Full Accessibility & 508 Audit: 
- [x] Product Manager: create issue and fill in feature name in the title and other bolded information appropriately
- [ ] Practice area reviewers create sub-issues attached to this one with feedback for your team on *feature name*
- [ ] **Requesting team** completes attached feedback tickets based on definition of done included in each practice area feedback ticket (may differ for each practice area, please refer to attached tickets for further instruction)
- [ ] All requesting team attendees complete brief [VSP Collaboration Cycle Feedback](https://adhoc.optimalworkshop.com/questions/20260uu8-0-0/questions/before) survey

## Artifacts - _please provide the following artifacts when you request a full review_
- Access information for the tool or feature in staging _**do not put staging credentials in your `va.gov-team` ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository**_
- Link to your accessibility test plan:
  - **TestRail** [TestRail VSP accessibility test plan template](https://dsvavsp.testrail.io/index.php?/suites/view/14&group_by=cases:section_id&group_order=asc) **OR**
  - **Markdown** [Github accessibility test plan template](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/accessibility-test-cases-template.md)

## Additional Notes
As part of this review, VSP will send your product to VA's 508 office for approval (as applicable)

**Comment by jenstrickland** [comment]
@luke-gcio - Has the team completed the accessibility test cases documentation? I recommend using the [checklist method](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/accessibility/vsa-accessibility-test-cases-checklist.md) to certify that the team completed the required accessibility testing. Please let me know if I may be of help to get this audit ready to go.

**Comment by luke-gcio** [comment]
Spoke to VSP - with the understanding that BDD is part of Disability Claims, they are ok with combining this when we get to BDD.

**Comment by jenstrickland** [comment]
👍  Is this ticket a request for the full accessibility audit now?

**Comment by luke-gcio** [comment]
We have combined 526 and BDD so we need to do a full audit for both at the same time (they are tied at the hip).  We are ready to begin!  I will go make sure the test accounts for staging are correct.

**Comment by jenstrickland** [comment]
It's okay to do this as a two-in-one. Thanks!

**Comment by jenstrickland** [comment]
@luke-gcio @Mottie 

Would y'all please add the URLs to access and the Users to test to the details above?

**Comment by Mottie** [comment]
BDD flow:

- Open the following URL in a new tab to reset the wizard that is shown on the introduction page:
  https://staging.va.gov/disability/file-disability-claim-form-21-526ez/introduction
- Wizard: For separated from military question, choose "No"
- Wizard: For anticipated date of release question, set date to `Mar` `1` `2021` (less than 90 days from today)
- Wizard: Use "File a Benefits Delivery at Discharge claim" button
- Intro page: Sign in to user 228, or if already signed in use green button to start the form
- Intent to file message: Use "continue" button
- Step 1 of 5: Veteran details. Use "continue" button
  - This would be a good place to use the **"Open save-in-progress menu"** popup and paste in the BDD prefill data; if you don't know what this is, please message me and I can walk you through it
- Step 1 of 5: Military service history.
  - An existing history card should be collapsed with an "Edit" button
  - A new card with `Mar` `1` `2021` data should be prefilled in the "Service end dates" of a "New service period" card.
  - Choose a "Branch of service" & enter a "Service start date" and use "Save" button, if not already prefilled
  - "Continue" to next page
- Step 1 of 5: Place of anticipated separation. Start typing and then choose an option in the dropdown, e.g. "Fort Dix", if not already prefilled
- Now, everything is the same as form 526 until you reach this URL: https://staging.va.gov/disability/file-disability-claim-form-21-526ez/supporting-evidence/service-treatment-records
  - This is the page where BDD users will need to upload their service treatment records.
- The remainder of the flow is the exact same as other 526 claim types.

**Comment by joshkimux** [comment]
Not sure if a ticket has already been written up for the content here. Tagging @peggygannon for awareness!
<img width="1133" alt="Screen Shot 2021-01-14 at 11 01 08 AM" src="https://user-images.githubusercontent.com/14154792/104616344-2c94eb80-5658-11eb-906f-982495de05e5.png">

**Comment by peggygannon** [comment]
Hi, @joshkimux Thank you...  This content above is outdated. I've added below a link to the content outline we had recommended to the team for the 526/BDD wizard work and implementation. Please let me know if you have any questions. 

https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/public-websites/how-to-apply-wizards/content-outlines/disability-form.md

cc @luke-gcio @johnhashva

**Comment by joshkimux** [comment]
Not an accessibility issue, but more of a usability one @Mottie @kevinstachura 
On the review page, if the user selects "no," they will lose all of their uploaded documents and will have to re-upload all of them if they re-select "yes"

https://user-images.githubusercontent.com/14154792/107548271-48ba7880-6b83-11eb-9601-57f5d4b529f1.mp4

**Comment by joshkimux** [comment]
Another usability issue here. Empty states do not have unique messaging:
![No-empty-state-disabilities](https://user-images.githubusercontent.com/14154792/107548811-f463c880-6b83-11eb-8c4c-b70c2dc92e47.png)

**Comment by joshkimux** [comment]
@1Copenut @luke-gcio  I've tentatively completed this full a11y audit 🎉 
Special thanks to @Mottie and @kevinstachura for answering all my questions and carrying the effort 🙇 

I've noted this as tentative as additional issues may emerge while working on these tickets. Furthermore, I couldn't provide clear recommendations on some of the tickets as I think they deserve more conversation and thought. @1Copenut would love your review on these to see if they have appropriate severity levels and recommended solutions 🙏

**Comment by Mottie** [comment]
Hmm, that summary page shouldn't ever be blank. It'll contain any selected rated disability and all new disabilities. This may happen if you reload the page? If not, please let me know which user and if you made selections/added new disabilities. And if this happened after continuing a form, or you found it while stepping through the form in a single sitting. Thanks!

**Comment by Mottie** [comment]
That file upload problem you found... ouch... yeah that definitely needs to be fixed. I'll open a ticket.

**Comment by joshkimux** [comment]
@Mottie I'll check it again tomorrow! Fingers crossed it was just the browser being weird.
